### DISCUSSION TITLE
Generated alt text for result screenshots for sharing results in an inclusive manner

### DISCUSSION THREAD

**Discussion by meliache** [monkeytypegame/monkeytype#4761] [Ideas]
In Monkeytype I can generate screenshots of the test results for sharing on social media. On mastodon and the Fediverse in general it's common (and expected) to add [alt text](https://en.wikipedia.org/wiki/Alt_attribute) to images for visually impaired people.  In my alt text I summarized the statistics from the screenshot, but it would be cool if monkeytype could automatically generate it. Or is there a feature like this and I didn't find it?

**Comment by Miodec** [comment]
Isnt alt text completely separate to the image? Or is there a way to embed that inside the image?

**Comment by meliache** [comment]
> Isnt alt text completely separate to the image? Or is there a way to embed that inside the
> image?

Depends on what you refer to as "image". Afaik alt text is separate from the image *file*, but it can be embedded within the HTML image tag `<img alt="…">`.

But of course when we click "Copy screenshot to clipboard", it's the image *file*, not the HTML object, that's copied and I don't know an easy way to change that. But in my browser that button also opens the image file blob in the web-browser. Would it be possible to make that button open an HTML page which displays the image with an alt property, so that people who care can copy that? Maybe also just print the alt text as text caption under the image to make it more obvious, so that you don't need to use an inspector to copy that. Or maybe have a separate button to copy the statistics in text-form.

By the way, it also *is* possible to embed text within the header of images files. I've seen this with some stable-diffusion software which optionally embeds the prompt and hyperparameters used in generating an image in the image file. But I don't think that this is a good solution, as requires too much technical knowledge and effort and thur would not encourage people to use the alt text.

Ideally a solution should make using the alt text easy and encourage it. Therefore I think entail a change in the UI. I'm not a UI designer nor am I super knowledgeable in web-design or accessibility, so I'm not sure how exactly such a solution should be best implemented.

**Reply by Miodec** [reply]
Im not sure this is doable. Alt text is an html property, but at no point in the screenshot flow the image is displayed by HTML. It either goes directly to the clipboard or displayed directly by the browser itself.

And even if i displayed the image as HTML, using right click and copy image wouldnt copy the alt text i think.
